akka-actor/src/main/scala/akka/routing/Resizer.scala

```diff
@@ -273,11 +295,13 @@ private[akka] final class ResizablePoolCell(
       resizer.isTimeForResize(resizeCounter.getAndIncrement()) && resizeInProgress.compareAndSet(false, true)) {
       super.sendMessage(Envelope(ResizablePoolActor.Resize, self, system))
     }
+
     super.sendMessage(envelope)
   }
 
   private[akka] def resize(initial: Boolean): Unit = {
     if (resizeInProgress.get || initial) try {
+      tryReportMessageCount()
```

Comment:
I tried to avoid this as well. But I need the message counter to calculate the number of messages the router passed through during the period (since last resize). This can't be tracked in `resizer.resize` since it's not called on every message. It is possible to count the message in `resizer.isTimeForResize` as it's called every message but it's obviously a bad choice to add state mutation in a inquiry method.
That's why I couldn't think of a better design than the introduction of  tryReportMessageCount()